patroni/dcs/__init__.py

```diff
@@ -491,55 +517,62 @@ class Cluster(namedtuple('Cluster', 'initialize,config,leader,last_lsn,members,'
     :param history: reference to `TimelineHistory` object
     :param slots: state of permanent logical replication slots on the primary in the format: {"slot_name": int}
     :param failsafe: failsafe topology. Node is allowed to become the leader only if its name is found in this list.
-    :param workers: workers of the Citus cluster, optional. Format: {int(group): Cluster()}"""
-
-    def __new__(cls, *args):
-        # Make workers argument optional
-        if len(cls._fields) == len(args) + 1:
-            args = args + ({},)
-        return super(Cluster, cls).__new__(cls, *args)
+    :param workers: workers of the Citus cluster, optional. Format: {int(group): Cluster()}
+    """
+    initialize: Optional[str]
+    config: Optional[ClusterConfig]
+    leader: Optional[Leader]
+    last_lsn: int
+    members: List[Member]
+    failover: Optional[Failover]
+    sync: SyncState
+    history: Optional[TimelineHistory]
+    slots: Optional[Dict[str, int]]
+    failsafe: Optional[Dict[str, str]]
+    workers: Dict[int, 'Cluster'] = {}
 
     @staticmethod
-    def empty():
+    def empty() -> 'Cluster':
         return Cluster(None, None, None, 0, [], None, SyncState.empty(), None, None, None)
 
     @property
-    def leader_name(self):
+    def leader_name(self) -> Optional[str]:
         return self.leader and self.leader.name
 
-    def is_unlocked(self):
+    def is_unlocked(self) -> bool:
         return not self.leader_name
 
-    def has_member(self, member_name):
+    def has_member(self, member_name: str) -> bool:
         return any(m for m in self.members if m.name == member_name)
 
-    def get_member(self, member_name, fallback_to_leader=True):
+    def get_member(self, member_name: str, fallback_to_leader: bool = True) -> Union[Member, Leader, None]:
         return ([m for m in self.members if m.name == member_name] or [self.leader if fallback_to_leader else None])[0]
 
-    def get_clone_member(self, exclude):
-        exclude = [exclude] + [self.leader.name] if self.leader else []
+    def get_clone_member(self, exclude_name: str) -> Union[Member, Leader, None]:
+        exclude = [exclude_name] + [self.leader.name] if self.leader else []
```

Comment:
```suggestion
        exclude = [exclude_name] + ([self.leader.name] if self.leader else [])
```

It has nothing to do with this actual PR, but I was wondering if this check is missing parenthesis?